# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/xplanet/hotfix-giflib51.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

giflib 5.1 added an error code argument to [DE]GifCloseFile().

--- xplanet-1.3.1/src/libimage/gif.c	2026-09-27 13:22:46.998140738 +0200
+++ xplanet-1.3.1-fixed/src/libimage/gif.c	2026-09-27 13:22:47.008405102 +0200
@@ -24,6 +24,11 @@
 
 #include <gif_lib.h>
 
+#if GIFLIB_MAJOR > 5 || (GIFLIB_MAJOR == 5 && GIFLIB_MINOR >= 1)
+#define DGifCloseFile(f) DGifCloseFile(f, NULL)
+#define EGifCloseFile(f) EGifCloseFile(f, NULL)
+#endif
+
 /*
   Based on the gif2rgb and rgb2gif codes in the libgif distribution. 
 */
